binding/python/test_orca.py

```diff
@@ -0,0 +1,172 @@
+#
+#    Copyright 2024 Picovoice Inc.
+#
+#    You may not use this file except in compliance with the license. A copy of the license is located in the "LICENSE"
+#    file accompanying this source.
+#
+#    Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+#    an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+#    specific language governing permissions and limitations under the License.
+#
+
+import argparse
+import os
+import sys
+import unittest
+from typing import List
+
+import editdistance
+import pvleopard
+
+from _orca import Orca, OrcaError
+from _util import default_library_path, default_model_path
+from test_util import get_model_paths
+
+TEXT = "It doesn't matter how slowly you go, as long as you do not stop!"
+TEXT_NO_PUNCTUATION = "It doesn't matter how slowly you go as long as you do not stop"
+TEXT_CUSTOM_PRON = "I {live|L IH V} in {Sevilla|S EH V IY Y AH}. We have great {live|L AY V} sports!"
+
+
+class OrcaTestCase(unittest.TestCase):
+    access_key: str
+    wer_threshold: float
+    orcas: List[Orca]
+
+    @classmethod
+    def setUpClass(cls):
+        cls.orcas = [
+            Orca(
+                access_key=cls.access_key,
+                model_path=model_path,
+                library_path=default_library_path('../..'))
+            for model_path in get_model_paths()]
+
+    @classmethod
+    def tearDownClass(cls):
+        for orca in cls.orcas:
+            orca.delete()
+
+    def test_valid_punctuation_symbols(self) -> None:
+        for orca in self.orcas:
+            symbols = orca.valid_punctuation_symbols
+            self.assertGreaterEqual(len(symbols), 0)
+            self.assertTrue(all(isinstance(x, str) for x in symbols))
+            self.assertTrue("," in symbols)
+
+    def test_max_character_limit(self) -> None:
+        for orca in self.orcas:
+            self.assertGreaterEqual(orca.max_character_limit, 0)
+
+    def test_sample_rate(self) -> None:
+        for orca in self.orcas:
+            self.assertGreater(orca.sample_rate, 0)
+
+    def test_synthesize(self) -> None:
+        leopard = pvleopard.create(access_key=self.access_key)
+
+        for orca in self.orcas:
+            pcm = orca.synthesize(TEXT)
+            self.assertGreater(len(pcm), 0)
+
+            pcm = orca.synthesize(TEXT)
+
+            ground_truth = TEXT_NO_PUNCTUATION.split()
+            predicted, _ = leopard.process(pcm)
```

Comment:
Nice! The good thing about this test is that we can easily implement it in other SDKs so long as we have access to Leopard